### DISCUSSION TITLE
Support Tailwind v4?

### DISCUSSION THREAD

**Discussion by somethingxdev** [htmlstreamofficial/preline#518] [General]
is it possible to use preline with the new tailwind v4?

**Comment by nijkor** [comment]
try this:
```
@import "tailwindcss";
@source "node_modules/preline/dist";
```

must work

**Comment by robinsonjohn** [comment]
I'm also curious when support for Tailwind v4 might be coming. I understand it is still in beta, but is there a roadmap for when this may roll out?

**Comment by BALOTIAS** [comment]
If you’re looking to get PrelineUI up and running with Tailwind CSS v4, I’ve got you covered (at least I hope so). As far as I can tell, the only thing standing in the way was an update to PrelineUI’s Tailwind plugin. So, I took a crack at it — check out my PR here:
https://github.com/htmlstreamofficial/preline/pull/554

In short, use my PR as the preline package and adapt the new css config like this:

**package.json**
```json
  ....
  "dependencies": {
    ...
    "preline": "github:BALOTIAS/preline-tw-v4",
    "tailwindcss": "^4.0.0-beta.9"
  }
```

**CSS**
```css
@import "tailwindcss";

/* Adds preline source files and plugin as described in the old tw quick setup example,
   only updated to the new tw v4 syntax: https://preline.co/docs/index.html */
@source "preline/dist"; /* <- not sure if it should point to the directory instead: "../preline/dist" */
@plugin "preline/plugin"; /* Uses newish shorthand syntax from https://github.com/BALOTIAS/preline-tw-v4 */

```

Here is the example repo of the PR: https://github.com/BALOTIAS/preline-tw-v4-example

Give it a whirl and let me know how it goes! <:9

**Reply by porobertdev** [reply]
can confirm that it works (tested with Astro 5.3).

**Reply by AnassL** [reply]
Can confirm, worked with Next.js 15

**Comment by hallelujah** [comment]
Any update on tailwind v4 support?

@BALOTIAS maybe you could make a PR with your patch?

**Reply by BALOTIAS** [reply]
Hey, so I had a [PR](https://github.com/htmlstreamofficial/preline/pull/554) open, but the maintainers said that they'll close it as they plan to get rid of the plugin file with the TW4 update, see: https://github.com/htmlstreamofficial/preline/pull/554#issuecomment-2660942492

**Comment by jahaganiev** [comment]
Full Tailwind v4 support is live!

**Reply by duckeydev** [reply]
> Full Tailwind v4 support is live!

so uhm, the docs has issues with sidebar! and by the way
```ps
PS C:\Users\Ducky\cloudable\Blog> bun run build
$ bunx @tailwindcss/cli -i ./assets/css/style.css -o ./assets/css/tailwind.css --watch
Error: Can't resolve 'preline/plugin' in 'C:\Users\Ducky\cloudable\Blog\assets\css'
error: script "build" exited with code ```

**Reply by eyuramn** [reply]
> Full Tailwind v4 support is live!

Nice! Where can I find upgrade documentation?

**Reply by duckeydev** [reply]
```npx @tailwindcss/upgrade```
